clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -96,6 +98,11 @@
     private final Metrics metrics;
     private final long defaultApiTimeoutMs;
 
+    private final ConcurrentHashMap.KeySetView<CompletableFuture<?>, Boolean> activeFutures =
```

Comment:
Is there ever a case where we have multiple futures waited on? In other words, could this be a simpler type, such as `AtomicReference`?

Reply:
yeah i think so - i'll change it to atomic reference